docs/hsi.xml

```diff
@@ -0,0 +1,875 @@
+      <para>
+        This security level corresponds to the most basic of security protections
+        considered essential by security professionals.
+        Any failures at this level would have critical security impact.
+      </para>
+    </refsect2>
+
+    <refsect2 id="hsi-level2">
+      <title>HSI:3 (Theoretical)</title>
+      <para>
+        This security level corresponds to firmware security issues that pose a
+        theoretical concern or where any exploit would be difficult or
+        impractical to use.
+        At this level various technologies may be employed to protect the boot
+        process from modification by an attacker with local access to the machine.
+      </para>
+    </refsect2>
+
+    <refsect2 id="hsi-level4">
+      <title>HSI:4 (System Protection)</title>
+      <para>
+        This security level corresponds to out-of-band protection of the system
+        firmware perhaps including recovery.
+      </para>
+    </refsect2>
+
+    <refsect2 id="hsi-level5">
+      <title>HSI:5 (System Attestation)</title>
+      <para>
+        This security level corresponds to out-of-band attestation of the system
+        firmware.
+        There are currently no tests implemented for HSI:5 and so this security
+        level cannot yet be obtained.
+      </para>
+    </refsect2>
+
+    <refsect3 id="runtime-u">
+      <title>HSI Runtime Suffix <code>U</code></title>
+      <para>
+        Updates available on the <ulink url="https://fwupd.org/">
+        Linux Vendor Firmware Service </ulink> which are less than 12 months old.
+      </para>
+    </refsect3>
+
+    <refsect3 id="runtime-a">
+      <title>HSI Runtime Suffix <code>A</code></title>
+      <para>
+        Attestation data is available to verify the current PCR0 TPM hash.
+      </para>
+    </refsect3>
+
+    <refsect3 id="runtime-bang">
+      <title>HSI Runtime Suffix <code>!</code></title>
+      <para>
+        A runtime security issue detected.
+      </para>
+      <itemizedlist>
+        <listitem>
+          <para>
+            UEFI <ulink url="https://wiki.ubuntu.com/UEFI/SecureBoot">
+            Secure Boot</ulink> has been turned off. <emphasis>v1.5.0</emphasis>
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            The kernel is <ulink url="https://www.kernel.org/doc/html/latest/admin-guide/tainted-kernels.html">
+            tainted</ulink> due to a non-free module or critical firmware issue. <emphasis>v1.5.0</emphasis>
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            The kernel is not not <ulink url="https://mjg59.dreamwidth.org/50577.html">
+            locked down</ulink>. <emphasis>v1.5.0</emphasis>
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            Unencrypted <ulink url="https://wiki.archlinux.org/index.php/Dm-crypt/Swap_encryption">
+            swap partition</ulink>. <emphasis>v1.5.0</emphasis>
```

Comment:
Maybe worth including a link to your blog for this too?

Reply:
I considered it, but I wanted an independent source really. Like a wikipedia page, if it existed.

Reply:
OK.